src/Azure/Orleans.Transactions.AzureStorage/TransactionalState/StateEntity.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.WindowsAzure.Storage.Table;
+using Newtonsoft.Json;
+using Orleans.Transactions.Abstractions;
+
+namespace Orleans.Transactions.AzureStorage
+{
+    internal class StateEntity : TableEntity
+    {
+        public const string RKMin = "ts_";
+        public const string RKMax = "ts_~";
+
+        public string TransactionId { get; set; }
+        public long SequenceId { get; set; }
+        public string StateJson { get; set; }
+
+        public static StateEntity Create<T>(JsonSerializerSettings JsonSettings,
+            string partitionKey, PendingTransactionState<T> pendingState)
+            where T : class, new()
+        {
+            return new StateEntity
+            {
+                PartitionKey = partitionKey,
+                RowKey = MakeRowKey(pendingState.TransactionId, pendingState.SequenceId),
+                TransactionId = pendingState.TransactionId,
+                SequenceId = pendingState.SequenceId,
+                StateJson = JsonConvert.SerializeObject(pendingState.State, JsonSettings)
```

Comment:
Not robust if state is larger than what can be stored in Azure table row. Again, maybe o.k. if there is a warning somewhere against using this provider in production.

Reply:
Understood.  The plan is to define expected behaviors (including state size and other limitations) as part of the API definition.  We've not done that yet.  Will add tracking issue.